lib/help_scout/api/access_token.rb

```diff
@@ -1,32 +1,24 @@
 # frozen_string_literal: true
 
+require 'date'
+require 'help_scout/api/access_token/cache'
+require 'help_scout/api/access_token/request'
+
 module HelpScout
   class API
     class AccessToken
       class << self
         def create
-          connection = HelpScout::API::Client.new(authorize: false).connection
-          response = connection.post('oauth2/token', token_request_params)
-
-          case response.status
-          when 200 then new HelpScout::Response.new(response).body
-          when 429 then raise HelpScout::API::ThrottleLimitReached, response.body&.dig('error')
-          else raise HelpScout::API::InternalError, "unexpected response (status #{response.status})"
-          end
-        end
+          cache = HelpScout::API::AccessToken::Cache.new
+          request = HelpScout::API::AccessToken::Request.new
 
-        def refresh!
-          HelpScout.api.access_token = create
+          cache.configured? ? cache.fetch_token { request.execute } : request.execute
         end
 
-        private
+        def refresh!
```

Comment:
That `refresh!` method looks like it needs some general refactoring. Like it should probably be on the top level module since it's actually changing `HelpScout.api.access_token`, but I think that's outside the scope of this PR.

One small step in that direction would be moving the `refresh!` call into `HelpScout::API::Client#access_token`, but that's not critical.

Reply:
> Not sure about `refresh!` sometimes returning an existing access_token.

I think @jcypret's point (which I agreed with) was that a consistent return value (a token) was preferable to a situational one (`nil` on a no-refresh-needed call, a token when a refresh actually occurred).

I can see the argument that it could be confusing if a token isn't actually refreshed, maybe that indicates that we shouldn't be calling a `refresh!`-style method outside of the class.